File: mapeable/Mapeable/Manejo de propiedades/BuscadorDePropiedadesLegibles.vb
Namespace ManejoDePropiedades
    Public Class BuscadorDePropiedadesLegibles
        Private elTipo As Type
        Private lasPropiedades As IEnumerable(Of Propiedad)
        Private lasPropiedadesLegibles As IEnumerable(Of Propiedad)

        Public Sub New(elTipo As Type)
            Me.elTipo = elTipo
        End Sub

        Public Function EncuentreLasPropiedadesLegibles() As IEnumerable(Of Propiedad)
            ListeLasPropiedades()
            EncuentreLasLegibles()

            Return lasPropiedadesLegibles
        End Function

        Private Sub ListeLasPropiedades()
            Dim elBuscador As New BuscadorDePropiedadesPublicas(elTipo)
            lasPropiedades = elBuscador.EncuentreLasPropiedadesPublicas()
        End Sub

        Private Sub EncuentreLasLegibles()
            lasPropiedadesLegibles = lasPropiedades.Where(Function(c) c.SePuedeLeer)
        End Sub
    End Class
End Namespace

File: mapeable/Mapeable.ComparacionesParaPruebasUnitarias/Manejo de propiedades/BuscadorDePropiedadesLegibles.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;

namespace ManejoDePropiedades
{
    public class BuscadorDePropiedadesLegibles
    {
        private Type elTipo;
        private IEnumerable<Propiedad> lasPropiedades;

        private IEnumerable<Propiedad> lasPropiedadesLegibles;
        public BuscadorDePropiedadesLegibles(Type elTipo)
        {
            this.elTipo = elTipo;
        }

        public IEnumerable<Propiedad> EncuentreLasPropiedadesLegibles()
        {
            ListeLasPropiedades();
            EncuentreLasLegibles();

            return lasPropiedadesLegibles;
        }

        private void ListeLasPropiedades()
        {
            BuscadorDePropiedadesPublicas elBuscador = new BuscadorDePropiedadesPublicas(elTipo);
            lasPropiedades = elBuscador.EncuentreLasPropiedadesPublicas();
        }

        private void EncuentreLasLegibles()
        {
            lasPropiedadesLegibles = lasPropiedades.Where(c => c.SePuedeLeer);
        }
    }
}